mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/vault/attachements/AttachmentsFragment.kt

```diff
@@ -733,15 +733,32 @@ class AttachmentsFragment :
     }
 
     private fun deleteFiles(vaultFiles: List<VaultFile?>) {
-        if (vaultFiles.isEmpty()) return
+        // Check if the input list is empty, if so, return early
+        if (vaultFiles.isEmpty()) {
+            return
+        }
 
+        // If there are multiple items in the list, delete selected media files
         if (vaultFiles.size > 1) {
-            viewModel.deleteVaultFiles(attachmentsAdapter.selectedMediaFiles)
+            val selectedFiles = attachmentsAdapter.selectedMediaFiles
+            viewModel.deleteVaultFiles(selectedFiles)
         } else {
-            vaultFiles[0]?.let { viewModel.deleteVaultFile(it) }
+            // Retrieve the single item from the list
+            val singleVaultFile = vaultFiles[0]
+
+            // Check if the item is a directory, delete all files inside the directory
+            if (singleVaultFile?.type == VaultFile.Type.DIRECTORY) {
+                viewModel.deleteVaultFiles(vaultFiles)
```

Comment:
At this point vaultFiles is just a directory, the files it contains will not be deleted.